document-store/src/main/java/org/hypertrace/core/documentstore/model/config/mongo/MongoConnectionConfig.java

```diff
@@ -59,7 +59,10 @@ public MongoConnectionConfig(
 
   public MongoClientSettings toSettings() {
     final MongoClientSettings.Builder settingsBuilder =
-        MongoClientSettings.builder().applicationName(applicationName()).retryWrites(true);
+        MongoClientSettings.builder()
+            .applicationName(applicationName())
+            .retryWrites(true)
+            .retryReads(true);
```

Comment:
What's the default retry count (and backoff strategy)? Any impact on mongo client read timeout?

Reply:
This is the default supported by Mongo driver.
According to the documentation @ https://www.mongodb.com/docs/manual/core/retryable-reads/,
only one retry will be performed.